### ISSUE TITLE
[BUG]: MockResponsePlugin only returns the last "Set-Cookie" headers when returning the response

### ISSUE THREAD

**Issue by schalkvanwyk** [microsoft/dev-proxy#927]
### Description

When a mock file contains a request response with multiple headers having the same name, then only the last header will be added to the response headers.

I.e.: having multiple "Set-Cookie" headers (which is normal) results in only the last "Set-Cookie" header being added to the response headers collection.

### Expected behaviour

Every "Set-Cookie" response header should be sent with the response.

### Actual behaviour

Multiple "Set-Cookie" headers are defined in the mock response file, but only the last "Set-Cookie" is set in the response headers.

### Steps to reproduce

Create a mock response that has multiple "" defined in the response headers:

```
"response": {
"headers":`` [
          {
            "name": "Set-Cookie",
            "value": "XSRF-TOKEN=1234567890abcdefg; path=/; SameSite=Lax"
          },
          {
            "name": "Set-Cookie",
            "value": "MyCookie=Blah; path=/; expires=Thu, 31-Oct-2049 22:18:03 GMT; HttpOnly; SameSite=Lax"
          }
]
}
```

### Dev Proxy Version

0.21.0

### Operating system (environment)

Windows

### Shell

cmd

### Configuration file

{
  "$schema": "https://raw.githubusercontent.com/microsoft/dev-proxy/main/schemas/v0.21.0/rc.schema.json",
  "plugins": [
    {
      "name": "ExecutionSummaryPlugin",
      "enabled": true,
      "pluginPath": "~appFolder/plugins/dev-proxy-plugins.dll"
    },
    {
      "name": "MockGeneratorPlugin",
      "enabled": true,
      "pluginPath": "~appFolder/plugins/dev-proxy-plugins.dll"
    },
    {
      "name": "MockResponsePlugin",
      "enabled": true,
      "pluginPath": "~appFolder/plugins/dev-proxy-plugins.dll",
      "configSection": "mocksPlugin"
    }
  ],
  "urlsToWatch": [
    "https://jsonplaceholder.typicode.com/*",
    "https://online.sage.co.za/U04992/*"
  ],
  "mocksPlugin": {
    "blockUnmockedRequests": true
  },
  "genericRandomErrorPlugin": {
    "errorsFile": "devproxy-errors.json"
  },
  "logLevel": "information",
  "newVersionNotification": "stable"
}

**Comment by schalkvanwyk** [comment]
Maybe... Also consider being able to add an array of values in the definition of a response header.

"response": {
          "headers":`` [
          {
            "name": "Set-Cookie",
            "value": [
                        "XSRF-TOKEN=1234567890abcdefg; path=/; SameSite=Lax",
                        "MyCookie=Blah; path=/; expires=Thu, 31-Oct-2049 22:18:03 GMT; HttpOnly; SameSite=Lax"
            ]
          }
}

**Comment by schalkvanwyk** [comment]
Another update... 

I found the following MSDN reference: [CookieContainer.SetCookies](https://learn.microsoft.com/en-us/dotnet/api/system.net.cookiecontainer.setcookies?view=net-8.0#remarks)
"_The HTTP cookies in the cookieHeader string must be delimited by commas._"

I tried the above and it seems to work with the "MockResponsePlugin". So the following results in both cookies being sent:
```
"response": {
          "headers":`` [
          {
            "name": "Set-Cookie",
            "value": "XSRF-TOKEN=1234567890abcdefg; path=/; SameSite=Lax;, MyCookie=Blah; path=/; expires=Thu, 31-Oct-2049 22:18:03 GMT; HttpOnly; SameSite=Lax;"
          }
}
```

Just as a note, I don't see the same definition in the MDN docs for [Set-Cookie](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie), it only mentions:
"_To send multiple cookies, multiple Set-Cookie headers should be sent in the same response._"

So the issue might be that the mock recorder does not create the headers with comma separators (as required by dotNet CookiesContainer), instead it outputs multiple headers as per the MDN docs. Seems there are just too many variations to cater for...

**Comment by waldekmastykarz** [comment]
Sorry for the trouble and thanks for the additional research. Let us have a look if we can make it easier. The reason why we moved away from a dictionary to an array was exactly to support multiple headers with the same name. Let us check if there's something special about cookies or if there's something else going on that prevents this from working as expected.

**Comment by waldekmastykarz** [comment]
I can confirm the issue and it's something of our own doing: we have a logic to merge headers from the mock response with existing headers here: https://github.com/microsoft/dev-proxy/blob/1334b3511a46f5ba576c6f06e44274b2b9090b85/dev-proxy-abstractions/ProxyUtils.cs#L328. Unfortunately, in the process we haven't considered multiple headers with the same name being set on the mock response. We'll get that fixed and released in beta for you to test asap. Thanks for bringing it up and sorry for the trouble.

**Comment by waldekmastykarz** [comment]
We've just released a beta version (v0.23.0-beta.1) with the issue fixed. It's available on Homebrew already and should become available on winget later today/tomorrow.
